src/main/java/io/divolte/server/filesinks/gcs/GoogleCloudStorageFileManager.java

```
@@ -0,0 +1,406 @@
+                                new SourceObject(inflightPartialName)));
+
+                final URL composeUrl = composeUrlFor(bucketEncoded, composeDestinationObjectEncoded);
+                final GcsObjectResponse composeResponse = googlePost(composeUrl, GcsObjectResponse.class, JSON_CONTENT_TYPE_HEADER, os -> {
+                    MAPPER.writeValue(os, composeRequest);
+                });
+
+                logger.debug("Google Cloud Storage compose response {}", composeResponse);
+            }
+        }
+    }
+
+    public static class GoogleCloudStorageFileManagerFactory implements FileManagerFactory {
+        private final Schema schema;
+        private final ValidatedConfiguration configuration;
+        private final String name;
+
+        private GoogleCloudStorageFileManagerFactory(final ValidatedConfiguration vc, final String sinkName, final Schema schema) {
+            this.schema = Objects.requireNonNull(schema);
+            this.configuration = Objects.requireNonNull(vc);
+            this.name = Objects.requireNonNull(sinkName);
+        }
+
+        @Override
+        public void verifyFileSystemConfiguration() {
+            // Just perform a get on the bucket for access verification
+            try {
+                /*
+                 * Get the credentials. This is a redundant operation, just to provide a nicer
+                 * error message if the presence of default credentials are the issue instead of
+                 * the actual connection / ACLs / bucket existence.
+                 */
+                getGoogleCredentials();
+            } catch (final IOException ioe) {
+                logger.error("Failed to obtain application default credentials for Google Cloud Storage for OAuth scope '" + GoogleCloudStorageFileManager.GCS_OAUTH_SCOPE + "'", ioe);
+                throw new RuntimeException("Could not obtain application default credentials for Google Cloud Storage", ioe);
```

Changed it. Note that this has always been a bare `RuntimeException`. I think all other sections where we want to fail hard on broken config use `RuntimeException` as well.